File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_index_map = {}

        for current_index, current_num in enumerate(nums):
            complement = target - current_num

            if complement in num_index_map:
                return [current_index, num_index_map[complement]]

            num_index_map[current_num] = current_index

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const numIndexMap = new Map();
    
    for (let currentIndex = 0; currentIndex < nums.length; currentIndex++) {
        const currentNum = nums[currentIndex];
        const difference = target - currentNum;

        if (numIndexMap.has(difference)) {
            return [currentIndex, numIndexMap.get(difference)];
        }
        
        numIndexMap.set(currentNum, currentIndex);
    }
};
